src/internal/user_api.rs

```diff
@@ -915,4 +953,14 @@ pub(crate) mod tests {
         let jwt = Jwt::try_from("eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlEwWXhNekUwTlVJeE9UVTNRakZFTlRZM01rVkNRakE0UkVNMk1UTkZOVGRETVRBNE9EQTVNUSJ9.eyJodHRwOi8vaXJvbmNvcmUxL3BpZCI6MSwiaHR0cDovL2lyb25jb3JlMS9raWQiOjE4NTksImh0dHA6Ly9pcm9uY29yZTEvc2lkIjoiSXJvbkhpZGUiLCJodHRwOi8vaXJvbmNvcmUxL3VpZCI6ImJvYi53YWxsQGlyb25jb3JlbGFicy5jb20iLCJpc3MiOiJodHRwczovL2lyb25jb3JlbGFicy5hdXRoMC5jb20vIiwic3ViIjoiZ2l0aHVifDExMzY4MTIyIiwiYXVkIjoiaEdFTHh1QktENjRsdFM0Vk5hSXkybXpWd3RxZ0phNWYiLCJpYXQiOjE1OTMxMzAyNTUsImV4cCI6MTU5MzEzMzg1NX0.J9sPgSFjucLQHpGOsEJ3xJf66nNK6Rf1n-C4YTsqWjPGwHlA8qyY4YIfNhwAjSstwvx2ImUb-Rf2Ghjq_4gpnArVfzkqa2HN06p_kRvwlL_kJoKTP8fo9LSpceNAbv75S4_EzOAWHTTNzDVjriQ1sjZYCYuD9BBjCG7ie0vSATb9uE4BtE_fSrlRkXlEW_608PDajNpwcCzSC-rMcWa1vDCYEuk405MzxkMJIi65ghMs9AEi6QotEhimf1gbrSaJFyyAqVKBPwA5--z64cK1vSwsX3mO2bCWIKbqLgXXWU0zr7saP9jVeMKXXetBW5KHHjYKRZ6lY9CquhtsnjSxvQ");
         assert!(!jwt.is_ok())
     }
+
+    #[test]
+    fn valid_jwt_duplicate_aliases() {
+        // This is a JWT generated with the following claims:
+        // { "http://ironcore1/pid": 1, "http://ironcore1/kid": 1859, "http://ironcore1/sid": "IronHide", "sid": "IronHide",
+        //    "http://ironcore1/uid": "bob.wall@ironcorelabs.com", "iss": "https://ironcorelabs.auth0.com/",
+        //    "sub": "github|11368122", "aud": "hGELxuBKD64ltS4VNaIy2mzVwtqgJa5f", "iat": 1593130255, "exp": 1593133855 }
+        let jwt = Jwt::try_from("eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwOi8vaXJvbmNvcmUvcGlkIjoxLCJodHRwOi8vaXJvbmNvcmUva2lkIjoxODU5LCJodHRwOi8vaXJvbmNvcmUvc2lkIjoiSXJvbkhpZGUiLCJzaWQiOiJJcm9uSGlkZSIsImh0dHA6Ly9pcm9uY29yZS91aWQiOiJib2Iud2FsbEBpcm9uY29yZWxhYnMuY29tIiwiaXNzIjoiaHR0cHM6Ly9pcm9uY29yZWxhYnMuYXV0aDAuY29tLyIsInN1YiI6ImdpdGh1YnwxMTM2ODEyMiIsImF1ZCI6ImhHRUx4dUJLRDY0bHRTNFZOYUl5Mm16Vnd0cWdKYTVmIiwiaWF0IjoxNTkzMTMwMjU1LCJleHAiOjE1OTMxMzM4NTV9.Bp5aXzT0k0busaETapSaOUyrEq1zkW-iLCyVEFP7ngEOJY1eU3BAuuDsIZmpW53oVCcCrgGt43VRse6oYegoz_ZsR2Zn2GjQ5JMPRdTZxrV5rjATJeCl9LZwwLktur0ik5j1KLDSA_hAlp8oaHfS50gtxh5qfG3e42o72jvdBcC9cOT45e2xEVPD67x7Ijt2NKPG7HQOyvKILMFuQtH1M0ywo9yNwJe4xS5n_B8F_PYZLe-EElSQY-ydXukfn0k1sQxt8VhgHZRUnKgmKVej5Z6_OtvgKONATIKJSnv1aI7C6il2yG7Di-b5YZr8R2hej57adBeISVoUdbOpgBON4g");
```

Comment:
I think we should test all the duplicates.

Reply:
Done